kie-server-parent/kie-server-services/kie-server-services-openshift/src/main/java/org/kie/server/services/openshift/impl/storage/cloud/KieServerStateOpenShiftRepository.java

```diff
@@ -184,6 +187,46 @@ public KieServerState load(@NotNull String serverId) {
         return kieServerState;
     }
     
+    public String getAppName(String serverId, OpenShiftClient client) {
+        DeploymentConfig dc = getKieServerDC(serverId, client);
+        return dc != null ? dc.getMetadata().getLabels().get(CFG_MAP_LABEL_APP_NAME) : getAppNameByHostname(client);
+    }
+    
+    public String getAppNameByHostname(OpenShiftClient client) {
+        String hostname = System.getenv(ENV_HOSTNAME);
+        if (hostname == null || client.pods().withName(hostname).get() == null) {
```

Comment:
Just thinking, what conditions can lead into this branch. In what cases this branch applies (I mean the real usage on OpenShift)?

Reply:
@sutaakar It is for safe guarding NPE. For running on OpenShift, those should never be null. I'm thinking of removing it and keeping code simple. What do you think?

Reply:
@sutaakar Refactored code by leveraging Optional API for better readability and NPE avoidance.